connectors/aws/aws-bedrock/src/main/java/io/camunda/connector/aws/bedrock/model/ConverseData.java

```diff
@@ -116,39 +121,53 @@ public BedrockResponse execute(
                                 .maxTokens(this.maxTokens)
                                 .topP(this.topP)
                                 .build()));
-    String newMessage = converseResponse.output().message().content().getFirst().text();
-    this.messages.add(new PreviousMessage(newMessage, ConversationRole.ASSISTANT.name()));
-    return new ConverseWrapperResponse(this.messages, newMessage);
+
+    var responseMessage = converseResponse.output().message();
+    this.messagesHistory.add(messageMapper.mapToBedrockMessage(responseMessage));
+
+    return messagesHistory;
+  }
+
+  private BedrockMessage prepareNewBedrockMessage(BedrockContentMapper bedrockContentMapper) {
+    String user = ConversationRole.USER.toString();
+    List<BedrockContent> contentList =
+        new ArrayList<>(bedrockContentMapper.documentsToBedrockContent(this.newDocuments));
+    contentList.add(bedrockContentMapper.messageToBedrockContent(this.newMessage));
+
+    return new BedrockMessage(user, contentList);
+  }
+
+  private MessageMapper createMessageMapper() {
+    var bedrockContentMapper = new BedrockContentMapper(new DocumentMapper());
+    return new MessageMapper(bedrockContentMapper);
```

Comment:
So we are not returning the new Message and the history but only the history ? 
That requires changes in the documentation, and is it what is expected ?

Reply:
Yes and Yes ))
I described this in the documentation [pr](https://github.com/camunda/camunda-docs/pull/4795)